adafruit_lsm6ds/__init__.py

```diff
@@ -320,13 +305,10 @@ def gyro_range(self):
     def gyro_range(self, value):
         if not GyroRange.is_valid(value):
             raise AttributeError("range must be a `GyroRange`")
-        if value is GyroRange.RANGE_4000_DPS and not isinstance(self, ISM330DHCX):
-            raise AttributeError("4000 DPS is only available for ISM330DHCX")
-
         if value is GyroRange.RANGE_125_DPS:
             self._gyro_range_125dps = True
             self._gyro_range_4000dps = False
-        elif value is GyroRange.RANGE_4000_DPS:
+        elif value == 4000:
```

Comment:
Why undo this? Maybe have a top-level GyroRange and mask the gyro_range property in subclass that don't support 4000 so they error? Something like:

```python
@property
def gyro_range(self, value):
    if value == GyroRange.RANGE_4000_DPS:
        raise ValueError()
    super().gyro_range = value
```

I'm not positive that will work since it's a property. You may need a private internal method instead.

Reply:
@tannewt please take a look at my changes when you have a chance. I'm now relying on the non-existance of the 4k range in the non-ISM classes to handle itself, and moved the 4k bit management into the subclass where it belongs.